### PR TITLE
added height config for bamboo

### LINKED ISSUE #4669: Option in Config for Bamboo Max height, like there is for Cactus and Sugar Cane
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
It'd be nice to be able to set the max height of bamboo as well just like you can with cactus and reeds already: 

`max-growth-height:
      cactus: 3
      reeds: 3
      bamboo: 16
`
**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by Machine-Maker**
Closes #4669 

Possible change that may be needed, I could only find cases where the IBlockAccess object passed to the bonemeal method was infact an instance of world, and I could only find cases where the IBlockAccess object passed to either of the height check methods in BlockBamboo were also instances of World. so I think all the casts to World from IBlockAccess should be fine, but I'm not sure.

Also, maybe this should go in zach's patch (0010) where the cactus and reed configs are.

### CODE REVIEW

**Review by Proximyst — DISMISSED** [review]
Squash with the other patch you mentioned.

**Review by kashike — DISMISSED** [review]

File: Spigot-Server-Patches/0599-Configurable-bamboo-natural-growth-heights.patch [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     public int cactusMaxHeight;
+     public int reedMaxHeight;
++    public int bambooMaxHeight;
++    public int bambooMinHeight;
+     private void blockGrowthHeight() {
+         cactusMaxHeight = getInt("max-growth-height.cactus", 3);
+         reedMaxHeight = getInt("max-growth-height.reeds", 3);
+-        log("Max height for cactus growth " + cactusMaxHeight + ". Max height for reed growth " + reedMaxHeight);
++        bambooMaxHeight = getInt("max-growth-height.bamboo.max", 16);
++        bambooMinHeight = getInt("max-growth-height.bamboo.min", 10);
```
Comment (nicegamer7) [inline Spigot-Server-Patches/0599-Configurable-bamboo-natural-growth-heights.patch]: Shouldn't this be 11?

Comment (Machine-Maker) [reply Spigot-Server-Patches/0599-Configurable-bamboo-natural-growth-heights.patch]: I took the min value directly from [Billy's patch on Purpur](https://github.com/pl3xgaming/Purpur/blob/ver/1.16.3/patches/server/0082-Implement-bamboo-growth-settings.patch).

I guess it does default to 11 initially.

Comment (BillyGalbreath) [reply Spigot-Server-Patches/0599-Configurable-bamboo-natural-growth-heights.patch]: Yes, 11 is correct here. For some reason I changed `<` to `<=` and used 10 in my patch. I dont remember why now :3

**Comment by Machine-Maker** [comment]
Ok, this was squashed into the cactus/reed patch and the default min height was bumped from 10 -> 11

**Review by zachbr — DISMISSED** [review]

**Comment by MiniDigger** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Pull request merged**